### ISSUE TITLE
reserved_concurrent_executions for aws_lambda_function does not support 0 concurrency

### ISSUE THREAD

**Issue by austinbyers** [terraform-providers/terraform-provider-aws#3803]
### Actual Result
If `reserved_concurrent_executions = 0` for an `aws_lambda_function` in Terraform, reserve concurrency is disabled:

![screen shot 2018-03-15 at 6 09 58 pm](https://user-images.githubusercontent.com/3608925/37498561-42b500ec-287c-11e8-9e5e-c1eda261915b.png)

### Expected Result
However, 0 is actually a valid value with a specific meaning, which you can see via the console:

![screen shot 2018-03-15 at 6 10 07 pm](https://user-images.githubusercontent.com/3608925/37500491-0cfb0392-2887-11e8-9df8-0902cf59c38a.png)

Why would you ever set reserved concurrency to 0? From the [AWS documentation](https://docs.aws.amazon.com/lambda/latest/dg/concurrent-executions.html):

_If you need a function to stop processing any invocations, you can choose to set the concurrency to 0 and throttle all incoming executions._

This is useful for long-running jobs (example: https://github.com/airbnb/binaryalert/issues/46)

### Requested Change
Provide a way to distinguish between _disabled_ reserve concurrency and _no_ reserve concurrency. Not sure the best approach here - maybe a negative value or an empty string disables it?

### Versions
Terraform 0.11.3 with AWS provider 1.11.0

### Affected Resource(s)
- aws_lambda_function

### Terraform Configuration Files
```hcl
resource "aws_lambda_function" "function" {
  function_name = "${var.function_name}"
  runtime       = "python2.7"
  role          = "${aws_iam_role.role.arn}"
  handler       = "${var.handler}"
  s3_bucket     = "${var.source_bucket}"
  s3_key        = "${var.source_object_key}"

  // Should temporarily throttle all invocations
  reserved_concurrent_executions = 0
}
```

**Comment by Puneeth-n** [comment]
Missed it when I implemented it. @austinbyers Thanks for raising the issue. :)

**Comment by Puneeth-n** [comment]
I introduced `-1` as a way of telling terraform to remove concurrency limit.

**Comment by bflad** [comment]
Hi everyone 👋Thank you for your patience with this one. Support for disabling Lambda Function invocations with setting `reserved_concurrent_executions` to `0` has been merged and will release with version 2.0.0 of the Terraform AWS Provider, likely in the next week or two.
